iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -400,7 +428,14 @@ public override async Task<long> SendTwinPatchAsync(TwinCollection reportedPrope
             }
             finally
             {
-                _twinResponseCompletions.TryRemove(correlationId, out _);
+                if (_twinResponseCompletions.ContainsKey(correlationId))
+                {
+                    _twinResponseCompletions.TryRemove(correlationId, out _);
+                    if (_twinResponseTimeouts.ContainsKey(correlationId))
```

Comment:
I would attempt to remove the entry from the timeouts dictionary even if it wasn't present in the completions dictionary.